text/0670-aws-applicationsignals-enablement-l2.md

```diff
@@ -0,0 +1,633 @@
+
+1. The application language and the SDK version
+    1. To decide the right image and image tag of the init container.
+       E.g. `public.ecr.aws/aws-observability/adot-autoinstrumentation-$lang:$$adot_sdk_version`
+    2. To decide the language specific environment varirables to be injected into the application container.
+       See [Environment variables injected by Application Signals auto-instrumentation](#environment-variables-injected-by-application-signals-auto-instrumentation).
+2. The OS and the CPU architecture
+    1. To decide the image tag of CloudWatch Agent.
+       E.g. the linux tag is `latest` and the windows image tag is `latest-windowsservercore2022`.
+    2. To decide what value to be set in `CORECLR_PROFILER_PATH`  for .NET applications.
+       See [DOTNET Linux x64](#dotnet-linux-x64), [DOTNET Linux ARM64](#dotnet-linux-arm64) and [DOTNET Windows](#dotnet-windows)
+
+```js
+export enum InstrumentationLanguage {
+  JAVA, PYTHON, DOTNET, NODEJS
+}
+
+export enum Instrumentation {
+  // This is just an example, the actual provided versions will be different.
+  JAVA_1_32_6,
+  PYTHON_V0_7_0,
+  DOTNET_V1_4_0,
+  NODEJS_V0_5_0
+}
+
+export interface InstrumentationProps {
+  // The application language
+  readonly language: InstrumentationLanguage;
+
+  // The ADOT language SDK version to be used
+  readonly sdkVersion: Instrumentation;
+
+  // The OS and CPU architecture used by the Task, defaults to linux-x64
+  readonly runtimePlatform?: ecs.RuntimePlatform;
+}
+```
+
+**Note**
+The SDK version is the image tag from the following ECR repository:
+
+1. <https://gallery.ecr.aws/aws-observability/adot-autoinstrumentation-java>
+2. <https://gallery.ecr.aws/aws-observability/adot-autoinstrumentation-python>
+3. <https://gallery.ecr.aws/aws-observability/adot-autoinstrumentation-node>
+4. <https://gallery.ecr.aws/aws-observability/adot-autoinstrumentation-dotnet>
+
+##### 2.2. Validation
+
+ApplicationSignalsTaskDefinition will validate:
+
+1. That the CloudWatch agent daemon deployment mode (enableSidecar=false) can only be used on ECS on EC2.
+
+ApplicationSignalsTaskDefinition won't validate:
+
+1. The correctness of the SDK version configured.
+
+##### 2.3. Override Mechanism
+
+Additionally, the new construct provides overrides mechanism through
+[EnvironmentExtension](#environmentextension) and [CloudWatchAgentProps](#cloudwatchagentprops).
+
+### Is this a breaking change?
+
+No.
+
+### What alternative solutions did you consider?
+
+N/A
+
+### What are the drawbacks of this solution?
+
+N/A
+
+### What is the high-level project plan?
+
+* [ ]  Gather feedback on the RFC
+* [ ]  Get bar raiser to sign off on RFC
+* [ ]  Make pull request to aws-cdk repository
+* [ ]  Iterate and respond to PR feedback
+* [ ]  Merge new construct and related changes
+
+### Are there any open issues that need to be addressed later?
+
+## Appendix
+
+### Environment variables injected by Application Signals auto-instrumentation
```

Comment:
I believe you should remove this section since you already mentioned the environment variables in the top of the doc

Reply:
Removed.